frontend/public/components/utils/resource-log.tsx

```diff
@@ -453,82 +513,110 @@ export const ResourceLog: React.FC<ResourceLogProps> = ({
-            actionLinks={
-              <AlertActionLink onClick={() => setError(false)}>{t('public~Retry')}</AlertActionLink>
+        <div className={classNames('resource-log__alert-wrapper')}>
+          {error && (
+            <Alert
+              isInline
+              className="co-alert co-alert--margin-bottom-sm"
+              variant="danger"
+              title={t('public~An error occurred while retrieving the requested logs.')}
+              actionLinks={
+                <AlertActionLink onClick={() => setError(false)}>
+                  {t('public~Retry')}
+                </AlertActionLink>
+              }
+            />
+          )}
+          {stale && (
+            <Alert
+              isInline
+              className="co-alert co-alert--margin-bottom-sm"
+              variant="info"
+              title={t('public~The logs for this {{resourceKind}} may be stale.', {
+                resourceKind: resource.kind,
+              })}
+              actionLinks={
+                <AlertActionLink onClick={() => setStale(false)}>
+                  {t('public~Refresh')}
+                </AlertActionLink>
+              }
+            />
+          )}
+          {hasTruncated && (
+            <Alert
+              isInline
+              className="co-alert co-alert--margin-bottom-sm"
+              variant="warning"
+              title={t('public~Some lines have been abridged because they are exceptionally long.')}
+            >
+              <Trans ns="public" t={t}>
+                To view unabridged log content, you can either{' '}
+                <ExternalLink href={linkURL}>open the raw file in another window</ExternalLink> or{' '}
+                <a href={linkURL} download={`${resource.metadata.name}-${containerName}.log`}>
+                  download it
+                </a>
+                .
+              </Trans>
+            </Alert>
+          )}
+        </div>
+        <div className={classNames('resource-log__log-viewer-wrapper')}>
+          <LogViewer
+            header={
+              <div className="log-window__header">
+                <HeaderBanner lines={lines} bufferFull={bufferFull} />
+              </div>
             }
-          />
-        )}
-        {stale && (
-          <Alert
-            isInline
-            className="co-alert co-alert--margin-bottom-sm"
-            variant="info"
-            title={t('public~The logs for this {{resourceKind}} may be stale.', {
-              resourceKind: resource.kind,
-            })}
-            actionLinks={
-              <AlertActionLink onClick={() => setStale(false)}>
-                {t('public~Refresh')}
-              </AlertActionLink>
+            theme="dark"
+            data={content}
+            ref={logViewerRef}
+            scrollToRow={currentLine}
+            height={isFullscreen ? '100%' : 350}
+            isTextWrapped={wrapLines}
+            toolbar={
+              <LogControls
```

Comment:
You probably want a `useEffect` for `LogControls` here as well. I'm fairly sure this will unmount/remount every render.